blockchain-source/src/cardano/event.rs

```diff
@@ -77,21 +80,34 @@ impl<'de> serde::Deserialize<'de> for BlockEvent {
             where
                 E: serde::de::Error,
             {
-                let mut reader = cbored::Reader::new(buf);
+                decode_deserialized_block(buf).map_err(serde::de::Error::custom)
+            }
 
-                let raw_block: SerializedBlock = reader
-                    .decode()
-                    .context("invalid block")
-                    .map_err(serde::de::Error::custom)?;
+            fn visit_seq<A: serde::de::SeqAccess<'de>>(
+                self,
+                mut seq: A,
+            ) -> Result<Self::Value, A::Error> {
+                let mut buf: Vec<u8> = Vec::with_capacity(seq.size_hint().unwrap_or(0));
+                while let Some(byte) = seq.next_element()? {
+                    buf.push(byte);
+                }
 
-                BlockEvent::from_serialized_block(raw_block).map_err(serde::de::Error::custom)
+                decode_deserialized_block(&buf).map_err(serde::de::Error::custom)
             }
         }
 
         deserializer.deserialize_bytes(V)
     }
 }
 
+fn decode_deserialized_block(buf: &[u8]) -> anyhow::Result<BlockEvent> {
+    let mut reader = cbored::Reader::new(buf);
+
+    let raw_block: SerializedBlock = reader.decode().context("invalid block")?;
+
+    BlockEvent::from_serialized_block(raw_block)
+}
+
```

Comment:
pls add test for deserialize and serialize

Reply:
the lack of a public constructor in `SerializedBlock` was also annoying for testing, so I ended up replacing it for a `Vec<u8>` and using derive.

so now I don't think adding the test adds anything.

although, on a second thought, `Box<[u8]>` would be a better type in this case.